src/core/file_storage/file_system_manager.go

```diff
@@ -145,17 +153,17 @@ func (m *FileManager) ReadFile(fileInfo *FileInfo) ([]byte, error) {
 
 	data, err := m.Fs.ReadFile(path.Join(m.Root, fileInfo.NamespaceId, fileObj.ParentDir, fileObj.RelativeFilename))
 	if err != nil {
-		return make([]byte, 0), nil
+		return nil, err
```

Comment:
I don't think we actually want this to fail. This is a case of lazy file creation. The file doesn't actually exist on disk until someone writes to it. So when someone requests the contents of an empty file, we just return an empty byte slice.